### PR TITLE
Clarify and restructure documentation for the CSSPlugin

### PR DESCRIPTION
**Pull request by cellog**
This is not ready for commit simply because I have not been able to figure out what purpose the "bundle" option serves by reading the source or the original commit that introduced it (7fb526507f7d4bc700c4ba0e79376fdcc3e0f143)

### CODE REVIEW

**Comment by nchanged** [comment]
Thanks a nice doc! thanks so much!
Could you take a look at https://github.com/fuse-box/fuse-box/issues/177 this one?

**Comment by nchanged** [comment]
Let's try to clarify the specs! We need this transparent:) I will leave it open for improvements

**Comment by cellog** [comment]
This PR is ready to go.  Please review the docs for accuracy, just added in the docs for bundle based on the brand spanking new commit, and marked both write and serve as deprecated based on #177

File: docs/builtin-plugins.md [hunk 1]
```diff
@@ -4,22 +4,72 @@ Fusebox contains premade plugins, that should help you to get started.
...
+  fsbx.CSSPlugin({
+    minify: true
+  })
+]
+```
+
+#### bundle
+
+This option is used to combine all css files into one file.  Within the bundle, the files can still be referenced with their original
+names, but the loader will request the single file that was made from bundling the css files together
```
Comment (aretecode) [inline docs/builtin-plugins.md]: might want to mention cache here as well

File: docs/builtin-plugins.md [hunk 2]
```diff
@@ -4,22 +4,72 @@ Fusebox contains premade plugins, that should help you to get started.
...
+]
+```
+
+* This simple code will change any scss file to a css file - `./main.scss` becomes `build/main.css` (your [outFile](#out-file) folder + project path)
+* it also creates `main.css.map` if sourcemaps have been specified
+* it automatically appends the new CSS files to the head of the web page and serves them
+
+A more extensive example can be found [here](https://github.com/fuse-box/angular2-example)
+
+### Options
```
Comment (aretecode) [inline docs/builtin-plugins.md]: if you lowercased serving down below, should keep it consistent 🔠

File: docs/builtin-plugins.md [hunk 3]
```diff
@@ -4,22 +4,72 @@ Fusebox contains premade plugins, that should help you to get started.
...
+    fsbx.CSSPlugin({})
+]
+```
+
+### Chaining CSS plugins to transform SCSS, Sass, and others
+
+Fusebox's plugin system allows simple transformation of more complex CSS systems through plugin chaining.  For example, to transform
+Sass into CSS:
+
+```
```
Comment (aretecode) [inline docs/builtin-plugins.md]: should mark as js

File: docs/builtin-plugins.md (hunk 2 above)
Comment (cellog) [inline docs/builtin-plugins.md]: This is a heading.  All headings are upper-cased.  The things below are actual code options, and are lower-cased in the code.

File: docs/builtin-plugins.md (hunk 1 above)
Comment (cellog) [inline docs/builtin-plugins.md]: I don't know what specifically needs mentioning, but if you tell me, I'll add it

File: docs/builtin-plugins.md (hunk 3 above)
Comment (cellog) [inline docs/builtin-plugins.md]: marked for this and all the other code blocks (most had no marking, from the original doc)

**Comment by nchanged** [comment]
Just leave it here

```
CSSPlugin({ bundle: `bundle.css` })]
```

1) Groups files into bundle.css
2) Inlines them in a virtual file (bundled)

```
CSSPlugin({ bundle: `bundle.css`, outFile : "build/bundle.css" })]
```
1) Will group files
2) Writes to DISK into build folder, 2 files : build/bundle.css and build/bundle.css.map (if sourcemaps are available)
3) Injects files automatically 

```
CSSPlugin({ bundle: `bundle.css`, outFile : "build/bundle.css", inject : false })]
```
All the same but does not inject the file. By default `inject` it's true:

**Comment by nchanged** [comment]
`inject` is currently under development. You can try it yourself, the changes are in master! Feel free to suggest improvements.

**Comment by cellog** [comment]
Thanks! What is the css file being injected into?

**Comment by nchanged** [comment]
into the head. 
inject : true // it's true by default
```
<link rel="stylesheet" href="bundle.css">
```

**Comment by devmondo** [comment]
@nchanged what about injecting as `<style>` tag like we have now ?

**Comment by devmondo** [comment]
@nchanged what about injecting as `<style>` tag like we have now ?

**Comment by nchanged** [comment]
@devmondo it's coming with `inline` options (and it's true by default)

**Comment by devmondo** [comment]
@nchanged  perfecto !

**Comment by nchanged** [comment]
This tests are now final https://github.com/fuse-box/fuse-box/blob/master/test/css_plugin.js

**Pull request closed without merging**